### PR TITLE
Add support for stampede2-skx and stampede2-knl.

### PR DESCRIPTION
**Pull request by fischer-ncar**
Add support for stampede2-skx and stampede2-knl at TACC.

Test suite: scripts_regression_tests on cheyenne, SMS.f19-g17.stampede2-[skx,knl]_intel, 
                    ERS.f19_g17.A.stampede2-skx
Test baseline:
Test namelist changes:
Test status: bit for bit

Fixes [CIME Github issue #]

User interface changes?:

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

File: config/cesm/machines/config_machines.xml.stampede [hunk 1]
```diff
@@ -0,0 +1,2119 @@
+<?xml version="1.0"?>
```
Comment (jedwards4b) [inline config/cesm/machines/config_machines.xml.stampede]: Why is this file here?

File: config/cesm/machines/config_machines.xml [hunk 2]
```diff
@@ -1719,24 +1719,82 @@
...
+      </modules>
+    </module_system>
+    <environment_variables>
+      <env name="OMP_STACKSIZE">256M</env>
+     </environment_variables>
+  </machine>
 
+  <machine MACH="stampede2-knl">
+    <DESC>Intel Xeon Phi 7250 ("Knights Landing") , batch system is SLURM</DESC>
+    <NODENAME_REGEX>.*stampede2</NODENAME_REGEX>
```
Comment (jedwards4b) [inline config/cesm/machines/config_machines.xml]: can't have the same NODENAME_REGEX for both definitions - I would suggest removing this one which would make skylake the default and require a --machine argument for knl

File: config/cesm/machines/config_machines.xml [hunk 3]
```diff
@@ -1719,24 +1719,82 @@
...
+    <OS>LINUX</OS>
+    <COMPILERS>intel</COMPILERS>
+    <MPILIBS>impi,mvapich2</MPILIBS>
+    <CIME_OUTPUT_ROOT>$ENV{SCRATCH}</CIME_OUTPUT_ROOT>
+    <DIN_LOC_ROOT>/scratch/projects/xsede/CESM/inputdata</DIN_LOC_ROOT>
+    <DIN_LOC_ROOT_CLMFORC>/scratch/projects/xsede/CESM/inputdata/inputdata/lmwg</DIN_LOC_ROOT_CLMFORC>
```
Comment (jedwards4b) [inline config/cesm/machines/config_machines.xml]: take the extra 'inputdata' out of this name

File: config/cesm/machines/config_machines.xml (hunk 2 above)
Comment (fischer-ncar) [reply config/cesm/machines/config_machines.xml]: Not a problem.  I've removed it.

File: config/cesm/machines/config_machines.xml (hunk 3 above)
Comment (fischer-ncar) [reply config/cesm/machines/config_machines.xml]: Fixed.

File: config/cesm/machines/config_machines.xml.stampede (hunk 1 above)
Comment (fischer-ncar) [reply config/cesm/machines/config_machines.xml.stampede]: I accidentally added it.  It's removed now.
